bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,249 @@
+        }
+
+        return true;
+    }
+
+    public boolean commandClear() {
+        try {
+            if (sendGet("/result.cgi?xr") == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandLogin() throws IOException {
+        return sendGet("/ergetcfg.cgi?lu=" + ETHERRAIN_USERNAME + "&lp=" + password) != null;
+    }
+
+    public boolean commandLogout() {
+        try {
+            if (sendGet("/ergetcfg.cgi?m=o") == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send logout command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private List<String> sendGet(String command) throws IOException {
+        String url = "http://" + address + ":" + port + "/" + command;
+
+        ContentResponse response;
+
+        try {
+            response = httpClient.newRequest(url).timeout(HTTP_TIMEOUT, TimeUnit.SECONDS).send();
+            if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+                logger.warn("Etherrain return status other than HTTP_OK : {}", response.getStatus());
+                return null;
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.warn("Could not connect to Etherrain with exception: {}", e.getMessage());
+            return null;
+        }
+
+        List<String> rVal = new BufferedReader(new StringReader(response.getContentAsString())).lines()
+                .collect(Collectors.toList());
+
+        return rVal;
+    }
+
+    private static EtherRainUdpResponse updBroadcast() {
+        DatagramSocket c;
+
+        // Find the server using UDP broadcast
+        try {
+            // Open a random port to send the package
+            c = new DatagramSocket();
+            c.setSoTimeout(BROADCAST_TIMEOUT);
+            c.setBroadcast(true);
+
+            byte[] sendData = BROADCAST_DISCOVERY_MESSAGE.getBytes();
+
+            // Try the 255.255.255.255 first
+            DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length,
+                    InetAddress.getByName("255.255.255.255"), BROADCAST_DISCOVERY_PORT);
+            c.send(sendPacket);
+
+            // Wait for a response
+            byte[] recvBuf = new byte[15000];
+            DatagramPacket receivePacket;
+            try {
+                receivePacket = new DatagramPacket(recvBuf, recvBuf.length);
+                c.receive(receivePacket);
+            } catch (SocketTimeoutException e) {
+                c.close();
+                return null;
+            }
+
+            // Check if the message is correct
+            String message = new String(receivePacket.getData()).trim();
```

Comment:
Also consider handling a message handling itself in separate method from the actual datagram-logic.

Reply:
Not sure what you mean here. This is the only place where we do a UDP broadcast it's only a simple discovery protocol. Would prefer to leave it one place.